### ISSUE TITLE
GetPosition Timeout

### ISSUE THREAD

**Issue by Hauraky** [Baseflow/flutter-geolocator#422]
Hi,
I am using your API to get accurate coordinates when the user clicks a button.
This is my function:
```
class Coordinate {
  Future<Position> getCoordinate() async {
    final Geolocator _myGeo = Geolocator()..forceAndroidLocationManager;

    DateTime _start = DateTime.now();
    Position _positionLast = await getLastKnowCoordinate();

    Position _position = await _myGeo.getCurrentPosition(desiredAccuracy: LocationAccuracy.best)
        .timeout(Duration(seconds: 2), onTimeout: () {
      debugPrint("getCoordinate - Timeout");
      return _positionLast;
    });
    debugPrint("getCoordinate - Esecution Time: " +
        DateTime.now().difference(_start).toString());

    return _position;
  } 
```
When I call this function, the phone waits about 4 seconds before obtaining the precise coordinates, it usually goes in timeout and returns LastKnowCoordinate.

I am testing it on Blackview 6800 Pro with Android 8.
Is there any way to speed up the function?
Do I have to call this function in the background every tot time to avoid this timeout or prepare the GPS on smartphone to catch the satellite? (I would like to avoid problems with battery consumption)

Thanks in advance

**Comment by mvanbeusekom** [comment]
@Hauraky this issue should be solved in version 6.0.0.

**Comment by Hauraky** [comment]
Hi,
I'm update the API to last version geolocator: ^6.0.0+4, but to retrieve the getCurrentPosition it needs about 3.6 seconds.
Let me know if I can do some test to help you.
Thanks
